src/store/actionCreators.js

```diff
@@ -69,13 +71,60 @@ export const deletePet = petId => {
   };
 };
 
-export const registerUser = ({
-  firstName,
-  lastName,
-  email,
-  password,
-  role,
-}) => {
+export const selectPet = (petId) => {
+  return async function (dispatch) {
+    try {
+      let response = await axios.get(`/pets/${petId}`);
+
+      let requests = await axios.get(`/pets/${petId}/requests`);
+      dispatch({ type: SELECT_PET, payload: response.data });
+      dispatch({ type: LIST_REQUESTS, payload: requests.data });
+    } catch (e) {
+      dispatch({ type: ERROR, payload: e.response.data.error });
+    }
+  };
+};
+
+export const updateRequest = (petId, requestId, status) => {
+  return async function (dispatch) {
+    try {
+      let response = await axios.put(`/pets/${petId}/requests/${requestId}`, {
+        responseStatus: status,
+      });
+      dispatch({ type: UPDATE_REQUEST, payload: response.data });
+    } catch (e) {
+      dispatch({ type: ERROR, payload: e.response.data.error });
+    }
+  };
+};
+
+export const bulkReject = (petId, ids) => {
+  return function (dispatch) {
+    try {
+      ids.forEach(async (id) => {
```

Comment:
You can use this functionality in backend. You may think if there are a lot of pet's request (like 1 million) in client side page could be blank or waiting to finish this. Or the user could reload page when forEach is not finish.

Reply:
Thank you very much for your feedback. I changed the logic so this can be managed by the backend, and all the Front has to do is to update the Redux state. This change was done in the Backed in the PR https://github.com/Cortesjohnj/adogta-backend/pull/20, so it would be great if you can check that PR as well so it can be approved as well as this one (if you consider it).

Thank you!